addons/pos_self_order/static/src/router.js

```diff
@@ -0,0 +1,94 @@
+/** @odoo-module */
+
+import { Component, useState, useSubEnv, xml } from "@odoo/owl";
+import { escapeRegExp } from "@web/core/utils/strings";
+import { browser } from "@web/core/browser/browser";
+import { zip } from "@web/core/utils/arrays";
+
+function parseParams(matches, paramSpecs) {
+    return Object.fromEntries(
+        zip(matches, paramSpecs).map(([match, paramSpec]) => {
+            const { type, name } = paramSpec;
+            switch (type) {
+                case "int":
+                    return [name, parseInt(match)];
+                case "string":
+                    return [name, match];
+                default:
+                    throw new Error(`Unknown type ${type}`);
+            }
+        })
+    );
+}
+export class Router extends Component {
+    static template = xml`<t t-slot="{{state.activeSlot}}" t-props="state.slotProps"/>`;
+    setup() {
+        useSubEnv({ navigate: this.navigate.bind(this) });
+        this.state = useState({
+            activeSlot: "default",
+            slotProps: {},
+        });
+        window.addEventListener("self_order_navigate", () => {
+            this.matchURL();
+        });
+        // this is needed for the back button to work
+        window.addEventListener("popstate", (event) => {
+            event.preventDefault();
```

Comment:
According to the spec this event is not cancelable and has no default action so this is likely useless:
https://www.w3.org/TR/2011/WD-html5-20110113/history.html#history-traversal

> [Queue a task](https://www.w3.org/TR/2011/WD-html5-20110113/webappapis.html#queue-a-task) to fire a [popstate](https://www.w3.org/TR/2011/WD-html5-20110113/history.html#event-popstate) event at the [Window](https://www.w3.org/TR/2011/WD-html5-20110113/browsers.html#window) object of the [Document](https://www.w3.org/TR/2011/WD-html5-20110113/infrastructure.html#document), using the [PopStateEvent](https://www.w3.org/TR/2011/WD-html5-20110113/history.html#popstateevent) interface, with the [state](https://www.w3.org/TR/2011/WD-html5-20110113/history.html#dom-popstateevent-state) attribute set to the value of state. This event must bubble but not be cancelable and has no default action.

Reply:
ok, i removed that line